### ISSUE TITLE
prec@1 prec@3

### ISSUE THREAD

**Issue by Augus-smile** [bryanyzhu/two-stream-pytorch#33]
Hello. I trained the resnet152 model on my own dataset. The accuracy of the train phase is quite good. The prec@1 of the train phase is 86%. But in the validation phase, the display accuracy is very low, and prec@1 and prec@3 The difference is relatively large, the prec@1 and prec@3 in the validation phase are 54% and 76% respectively. What is the reason for this? Prec@3 and prec@1 represent what they are, look forward to your reply, thank you!

**Comment by bryanyzhu** [comment]
Prec@3 means as long as one of your top 3 predictions is correct, your prediction is correct. It is a loose metric, so it should be higher than Prec@1. This makes sense.

For your problem, the most common explanation would be overfitting. How large is your own dataset? For example, how many classes, how many videos for each class, etc. As you should know, resnet152 can overfit UCF101 (UCF101 has 101 classes, each class has more than 100 videos). So if your dataset is smaller than UCF101, overfitting should be the reason.  You can try different things, like choosing a smaller network, early stopping, turn off BN, add more data augmentations etc. 

There is also another possibility, which is the domain similarity. Maybe your training data and validation data look very different, so that your model can't generalize. 

In general, validation accuracy will be lower than training accuracy, this is expected. Take UCF101 as an example, the training accuracy could be 90+, the validation accuracy is 80+. There is a 10% gap there as well. But again, since datasets are different, there is no direct comparison can be made. Hope this helps.

**Comment by Augus-smile** [comment]
Thank you for your answer. My dataset is similar to the UCF101 dataset. There are 100 classes, each with 250 videos. I think it should be the reason for overfitting, I will adopt your suggestion and try it. Thank you very much!

**Comment by bryanyzhu** [comment]
No problem at all, good luck with your experiments. Close the issue for now.
